program-runtime/src/compute_budget.rs

```diff
@@ -261,18 +260,15 @@ impl ComputeBudget {
             self.heap_size = Some(bytes as usize);
         }
 
-        let compute_unit_limit = if default_units_per_instruction {
-            updated_compute_unit_limit.or_else(|| {
+        let compute_unit_limit = updated_compute_unit_limit
+            .or_else(|| {
                 Some(
                     num_non_compute_budget_instructions
                         .saturating_mul(DEFAULT_INSTRUCTION_COMPUTE_UNIT_LIMIT),
                 )
             })
-        } else {
-            updated_compute_unit_limit
-        }
-        .unwrap_or(MAX_COMPUTE_UNIT_LIMIT)
-        .min(MAX_COMPUTE_UNIT_LIMIT);
+            .unwrap_or(MAX_COMPUTE_UNIT_LIMIT)
```

Comment:
this doesn't seem necessary:
`updated_compute_unit_limit` is an `Option`.
if it's `Some` we want to use that value, otherwise we want to use: `num_non_compute_budget_instructions.saturating_mul(DEFAULT_INSTRUCTION_COMPUTE_UNIT_LIMIT)`

Replace the `or_else` with `unwrap_or_else` and remove this `unwrap_or`.

Reply:
Ah yes. Was only removing dead codes after activation. But after removed `else` at Ln 271, the `unwrap_or` is redundant, could be removed.